security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -85,7 +85,9 @@ func (c *CitadelClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]strin
 	if err := c.reconnectIfNeeded(); err != nil {
 		return nil, err
 	}
-	ctx := metadata.NewOutgoingContext(context.Background(), metadata.Pairs("ClusterID", c.opts.ClusterID))
+	metakv := []string{"ClusterID", c.opts.ClusterID}
```

Comment:
Currently this PR has no tests. It would be good to have tests of this feature.

Reply:
yes, plan to submit the istio-agent part first. And then will the part for isitod and will add tests for it together.